src/dagmc/DagMC.hpp

```diff
@@ -0,0 +1,460 @@
+   */
+  bool has_prop(EntityHandle eh, const std::string& prop);
+
+  /** Get a list of all unique values assigned to a named property on any entity
+   *
+   *  @param prop The canonical property name
+   *  @param return_list Output param, a list of unique strings that are set as values for this property
+   *  @return MB_TAG_NOT_FOUND if prop is invalid.  Otherwise return any errors from
+   *          MOAB, or MB_SUCCESS if succesful
+   */
+  ErrorCode get_all_prop_values(const std::string& prop, std::vector<std::string>& return_list);
+
+  /** Get a list of all entities which have a given property
+   *
+   *  @param prop The canonical property name
+   *  @param return_list Output param, a list of entity handles that have this property
+   *  @param dimension If nonzero, entities returned will be restricted to the given dimension,
+   *                   i.e. 2 for surfaces and 3 for volumes
+   *  @parm value If non-NULL, only entities for which the property takes on this value will be returned.
+   *  @return MB_TAG_NOT_FOUND if prop is invalid.  Otherwise return any errors from
+   *          MOAB, or MB_SUCCESS if succesful
+   */
+  ErrorCode entities_by_property(const std::string& prop, std::vector<EntityHandle>& return_list,
+                                 int dimension = 0, const std::string* value = NULL);
+
+  bool is_implicit_complement(EntityHandle volume);
+
+  /** get the tag for the "name" of a surface == global ID */
+  Tag name_tag() {return nameTag;}
+
+  /** Get the tag used to associate OBB trees with geometry in load_file(..).
+   * not sure what to do about the obb_tag, GTT has no concept of an obb_tag on EntitySets - PCS
+   */
+  Tag obb_tag() { return NULL; }
+  Tag geom_tag() { return GTT->get_geom_tag(); }
+  Tag id_tag() { return GTT->get_gid_tag(); }
+  Tag sense_tag() { return GTT->get_sense_tag(); }
+
+ private:
+  /** tokenize the metadata stored in group names - basically borrowed from ReadCGM.cpp */
+  void tokenize(const std::string& str,
+                std::vector<std::string>& tokens,
+                const char* delimiters = "_") const;
+
+  /** a common type within the property and group name functions */
+  typedef std::map<std::string, std::string> prop_map;
+
+  /** Store the name of a group in a string */
+  ErrorCode get_group_name(EntityHandle group_set, std::string& name);
+  /** Parse a group name into a set of key:value pairs */
+  ErrorCode parse_group_name(EntityHandle group_set, prop_map& result, const char* delimiters = "_");
+  /** Add a string value to a property tag for a given entity */
+  ErrorCode append_packed_string(Tag, EntityHandle, std::string&);
+  /** Convert a property tag's value on a handle to a list of strings */
+  ErrorCode unpack_packed_string(Tag tag, EntityHandle eh,
+                                 std::vector< std::string >& values);
+
+  std::vector<EntityHandle>& surf_handles() {return entHandles[2];}
```

Comment:
Readability: Would make it more readable to define `2`, `3`, and `4` as used here as `static const int` or `#DEFINE` types, with the identifiers indicating they are related to surfaces, volumes, and groups.

These can also be reused on line 425 below, when doing `assert` checks.